src/storage/names.ts

```diff
@@ -22,30 +22,80 @@
 import * as storageModule from './module';
 
 
+/**
+ * Paths and file names for Blocks Projects
+ *
+ * Files in a project are stored in a directory whose name is the project name. All files have
+ * the extension '.json' and contain JSON text.
+ *
+ * Project information is stored in a file called 'project.info.json'.
+ *
+ * Modules
+ * Files that contain blocks are called modules. During deployment, a Python file is created for
+ * each module. Each Python file contains a Python class definition.
+ * There are three types of modules:
+ *     robot
+ *     mechanism
+ *     opmode
+ * A project consists of:
+ *     one project information file named 'project.info.json'
+ *     one robot module, named 'Robot.robot.json'
+ *     zero or more mechanisms, with the extension '.mechanism.json'
+ *     zero or more opmodes, with the extension '.opmode.json'
+ *
+ * The file path of the project info file is <ProjectName>/project.info.json.
+ * The file path of a module is <ProjectName>/<ClassName>.<ModuleType>.json.
+ */
+
+// The class name of the Robot module that is created automatically when a new project is created.
 export const CLASS_NAME_ROBOT = 'Robot';
+
+// The class name of the OpMode module that is created automatically when a new project is created.
 export const CLASS_NAME_TELEOP = 'Teleop';
 
+// The extension of all JSON files is .json.
 export const JSON_FILE_EXTENSION = '.json';
+
+// The extension of a downloaded project is .blocks.
 export const UPLOAD_DOWNLOAD_FILE_EXTENSION = '.blocks';
 
+// The file name of the project info file.
+const PROJECT_INFO_FILE_NAME = 'project.info.json';
+
+// The file name of the project info file.
+const ROBOT_MODULE_FILE_NAME = 'Robot.robot.json';
```

Comment:
Is this necessary for ease of parsing the filename?   If not, I would really prefer just robot.json

Reply:
Yes, ease of parsing the module file name. All module file names are <ClassName>.<module type>.json.